balancer-js/src/modules/pools/factory/weighted/weighted.factory.spec.ts

```diff
@@ -76,15 +104,67 @@ describe('creating weighted pool', async () => {
           return isSameAddress(log.address, factoryAddress);
         })
         .map((log) => {
-          try {
-            return weightedPoolFactoryInterface.parseLog(log);
-          } catch (error) {
-            console.error(error);
-            return null;
-          }
+          return weightedPoolFactoryInterface.parseLog(log);
         })
         .find((parsedLog) => parsedLog?.name === 'PoolCreated');
+      if (poolCreationEvent) {
+        poolAddress = poolCreationEvent.args.pool;
+      }
       expect(!!poolCreationEvent).to.be.true;
+      return;
+    });
+    it('should init join a pool', async () => {
+      const signerAddress = await signer.getAddress();
+      const weightedPoolInterface = new Interface(WeightedPool__factory.abi);
+      const pool = new Contract(poolAddress, weightedPoolInterface, provider);
+      const poolId = await pool.getPoolId();
+
+      const initJoinParams = weightedPoolFactory.buildInitJoin({
+        joiner: signerAddress,
+        poolId,
+        poolAddress,
+        tokensIn: tokenAddresses,
+        amountsIn: [
+          parseFixed('2000', 6).toString(),
+          parseFixed('8000', 6).toString(),
+        ],
+      });
+
+      const iERC20 = [
+        'function approve(address,uint256) nonpayable',
+        'function balanceOf(address) view returns(uint)',
+      ];
+      const erc20 = new Contract(AddressZero, iERC20);
+      // Approve vault for seeder
+      await Promise.all(
+        Object.values(tokenAddresses).map((address) => {
+          return erc20
+            .attach(address)
+            .connect(signer)
+            .approve(initJoinParams.to, parseEther('10').toString(), {
+              gasLimit: 3000000,
+            });
+        })
+      );
+      const tx = await signer.sendTransaction({
+        to: initJoinParams.to,
+        data: initJoinParams.data,
+        gasLimit: 30000000,
+      });
+      await tx.wait();
+      const receipt: TransactionReceipt = await provider.getTransactionReceipt(
+        tx.hash
+      );
+      const vaultInterface = new Interface(Vault__factory.abi);
+      const poolInitJoinEvent: LogDescription | null | undefined = receipt.logs
+        .filter((log: Log) => {
+          return isSameAddress(log.address, initJoinParams.to);
+        })
+        .map((log) => {
+          return vaultInterface.parseLog(log);
+        })
+        .find((parsedLog) => parsedLog?.name === 'PoolBalanceChanged');
+      expect(!!poolInitJoinEvent).to.be.true;
```

Comment:
Can you add a test to check the pool balances match the expected amounts.

Reply:
Done!